src/pages/resultsView/ResultsViewPageHelpers.ts

```diff
@@ -82,15 +82,17 @@ export function substitutePhysicalStudiesForVirtualStudies(cancerStudyIds:string
 
 }
 
-export function populateSampleSpecificationsFromVirtualStudies(samplesSpecification:SamplesSpecificationElement[], virtualStudies:VirtualStudy[]){
+export function populateSampleSpecificationsFromVirtualStudies(samplesSpecifications:SamplesSpecificationElement[], virtualStudies:VirtualStudy[]){
 
     const virtualStudiesKeyedById = _.keyBy(virtualStudies,(virtualStudy)=>virtualStudy.id);
-
+    const samplesSpecificationsKeyedByStudyId = _.keyBy(samplesSpecifications,(spec)=>spec.studyId);
     // remove specs for virtual studies (since they mean nothing to api)
     // and then populate with ids
-    samplesSpecification = _.filter(samplesSpecification,(spec)=>!virtualStudiesKeyedById[spec.studyId]);
+    samplesSpecifications = _.filter(samplesSpecifications,(spec)=>!virtualStudiesKeyedById[spec.studyId]);
```

Comment:
@dippindots i actually want "sampleSpecifications".    the plural "s" is on specifications, not on samples.   "sample" is a descriptive word.  I know you didn't write the code from the beginning but I want to make sure I communicate proper pluralization rules for you for the future!

Reply:
Once we make that change, we can release this.

Reply:
Thanks, @alisman! That helps a lot, I will remember this in the future. The only reason I change it to `samplesSpecifications` is that it is an array of `SamplesSpecificationElement`: https://github.com/cBioPortal/cbioportal-frontend/pull/2516/files#diff-98feaa0c1f30ed49f15d832754badb45R85. And the `SamplesSpecificationElement` is the specification of the `sampleList`(list of samples), I don't know if we should just use `sampleSpecifications` in this circumstances.